workers/common/src/main/kotlin/common/env/GitCredentialsGenerator.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.workers.common.env
+
+import java.net.URI
+
+import org.eclipse.apoapsis.ortserver.model.CredentialsType
+import org.eclipse.apoapsis.ortserver.model.InfrastructureService
+import org.eclipse.apoapsis.ortserver.workers.common.env.definition.EnvironmentServiceDefinition
+
+import org.slf4j.LoggerFactory
+
+/**
+ * A specialized generator class to generate files with special credentials for Git.
+ *
+ * This generator is required only for corner cases. Normally, the Git CLI should be able to obtain credentials from
+ * the _.netrc_ file. However, in some special cases, this mechanism is not working, and authentication against the
+ * repository is only possible if the credentials are placed in a _.git-credentials_ file. This generator is able to
+ * produce such a file, together with a Git configuration file that references it. It only processes infrastructure
+ * services whose credentials types contain [CredentialsType.GIT].
+ */
+class GitCredentialsGenerator : EnvironmentConfigGenerator<EnvironmentServiceDefinition> {
```

Comment:
Maybe "GitCredentials*File*Generator"?

Reply:
The name is in-line with the other generator classes, for instance `MavenSettingsGenerator`, `NetRcGenerator`, `NpmRcGenerator`.